### ISSUE TITLE
Force all code paths to 'return'

### ISSUE THREAD

**Issue by duarteroso** [vlang/v#6677]
Version: 0.1.29

### Problem
It is currently possible to write a function with an exhaustive `match` that does not return outside its scope.
Although the match covers all cases, it should be better if such function did not compile.

This isn't breaking code, barely throwing warning, but it is throwing warnings that are an annoyance when looking for the real problem :) 

```v
enum Enum{
	a
	b
	c
}

fn get (e Enum) int {
 	match(e) {
		.a { return 0 }
		.b { return 1 }
		.c { return 2 }
	}
}
```
Perhaps `get` should not compile as there is no return outside the `match` block.

### C generated
```c
int main__get (main__Enum e) { // warning: control reaches end of non-void function
	main__Enum _t78 = (e);
	if (_t78 == main__Enum_a) {
		return 0;
	} else if (_t78 == main__Enum_b) {
		return 1;
	} else if (_t78 == main__Enum_c) {
		return 2;
	};
}
```

**Comment by ntrel** [comment]
If every match branch has a return, there is no code path after the match. If this issue is about a C warning then V could generate something like `assert(0);` to silence it. BTW aren't there other C warnings in generated code?

**Comment by medvednikov** [comment]
Yeah if this `match` had a `else { return ... }`, then it should be fine. The old backend detected that.

This `match` doesn't have `else` though, so it should result in an error.

**Comment by Gladear** [comment]
There is no reason for this to be an error, all enum cases are handled.
Actually, in generated code, the last condition `if (_t78 == main__Enum_c)` is useless, because the result will always be `true` (as it isn't any of the cases before). It can be removed, so only `else` is kept, that should silence the warning (and result in more performant code).

**Comment by spaceface777** [comment]
The C warning was fixed in #6758.
